src/test/java/tech/toparvion/analog/service/choice/ChoicesPropertiesChangesListenerTest.java

```diff
@@ -0,0 +1,163 @@
+package tech.toparvion.analog.service.choice;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.cloud.endpoint.RefreshEndpoint;
+import tech.toparvion.analog.service.choice.ChoicesAutoReloadConfiguration.FileWatcherProvider;
+
+import java.nio.file.Path;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.List;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_MODIFY;
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+/**
+ * @author Polyudov
+ * @since v0.14
+ */
+@SuppressWarnings("ResultOfMethodCallIgnored")
+class ChoicesPropertiesChangesListenerTest {
```

Comment:
Great work! All the tests have successfully passed on my machine from the very first run 😃 
Adding some `@DisplayName` descriptions will also bring much benefit.

Reply:
Done